ipi/utils/parsing.py

```diff
@@ -190,48 +309,703 @@ def read_trajectory(
                     property_name: data_list,
                 }
             else:
-                ret = read_file(
-                    format,
-                    file_handle,
-                    dimension=dimension,
-                    units=units,
-                    cell_units=cell_units,
-                )
 
-                frame = ase.Atoms(
-                    ret["atoms"].names,
-                    positions=ret["atoms"].q.reshape((-1, 3)) * bohr2angstrom,
-                    cell=ret["cell"].h.T * bohr2angstrom,
-                    pbc=True,
-                )
+                # The unit conversion will be performed here because
```

Comment:
I would avoid modifying `read_file` because it's used in many places in the code, but usually only to read positions-like files.
In these scripts it would be used to read velocities, forces, etc and it seemed not to work correctly in these case.

Reply:
I think it does work properly. See #409 to do this "in the right place"